src/monodroid/jni/helpers.hh

```diff
@@ -0,0 +1,57 @@
+#ifndef __HELPERS_HH
+#define __HELPERS_HH
+
+#include <cstdint>
+
+#include "java-interop-util.h"
+#include "platform-compat.hh"
+
+namespace xamarin::android
+{
+#define ADD_WITH_OVERFLOW_CHECK(__ret_type__, __a__, __b__) xamarin::android::Helpers::add_with_overflow_check<__ret_type__>(__FILE__, __LINE__, (__a__), (__b__))
+#define MULTIPLY_WITH_OVERFLOW_CHECK(__ret_type__, __a__, __b__) xamarin::android::Helpers::multiply_with_overflow_check<__ret_type__>(__FILE__, __LINE__, (__a__), (__b__))
+
+	class Helpers
+	{
+	public:
+		template<typename Ret, typename P1, typename P2>
+		static force_inline Ret add_with_overflow_check (const char *file, uint32_t line, P1 a, P2 b) noexcept
```

Comment:
Why a static member function instead of a global function?

Reply:
I like to keep as much as possible in the classes, that's the only reason.